server/controllers/userAuth.js

```diff
@@ -254,15 +255,14 @@ const recoveryPassword = (req, resp) => {
 
 const updatePassword = (req, resp) => {
   const { password: updatedPassword, passwordConfirmation } = req.body;
-  const { matches } = validator;
   const { token } = req.params;
 
   const sanitizedToken = sanitize(token);
-  const validationError = !matches(updatedPassword, /^[^\s]{4,32}$/);
-  const passwordsNoMatch =
-    _trim(updatedPassword) !== _trim(passwordConfirmation);
 
-  if (validationError || passwordsNoMatch) {
+  if (
+    !validatePasswordStructure(updatedPassword) ||
+    _trim(updatedPassword) !== _trim(passwordConfirmation)
```

Comment:
Please either trim the input all way up before any processing or do not trim at all and just throw validation error when it contains leading/trailing whitespace.